### PR TITLE
Upgrade base app to Rails 5/Ruby 2.3.1

### PR DESCRIPTION
**Pull request by buitranquyet**
Also fixed issue for`cancancan` with different`policies`, it was not working previously.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  postgresql: 9.3
+  code_climate:
+    repo_token: ENTER_CODECLIMATE_REPO_TOKEN
+services:
+  - redis-server
+script:
+  - bundle exec massa -V
+env:
+  - >
+    CODECLIMATE_REPO_TOKEN=ENTER_CODECLIMATE_REPO_TOKEN
```
Comment (ecoologic) [inline .travis.yml]: @buitranquyet, so, is this currently configured to run, or is it just a copy and paste?

File: .rubocop.yml [hunk 2]
```diff
@@ -11,30 +11,27 @@ AllCops:
...
-Rails/FindBy:
-  Enabled: false # TODO enable after upgrade to Rails 4+
-
 Lint/HandleExceptions:
   Exclude:
     - 'bin/*'
 
 Metrics/AbcSize:
-  Max: 69 # TODO: change it to 25
+  Max: 15
```
Comment (ecoologic) [inline .rubocop.yml]: good luck writing 10 lines with ABC < 15... How about we just delete it and go with the default? (20 If I'm not mistaken)

Comment (buitranquyet) [inline .rubocop.yml]: @ecoologic the default is 15, check it here
https://github.com/bbatsov/rubocop/blob/master/config/default.yml#L1100

But yeah, 15 is too small, sometime very annoying, recently I just need to update edward to 18  https://github.com/net-engine/edward/blob/master/.rubocop.yml#L29

File: .travis.yml (hunk 1 above)
Comment (buitranquyet) [inline .travis.yml]: @ecoologic we just need to replace `ENTER_CODECLIMATE_REPO_TOKEN` with the correct value

**Pull request merged**